### PR TITLE
Make services decorator shortcuts explicit - closes #540

### LINKED ISSUE #540: Instance of 'Service' has no 'post' member (no-member)
Code uses magic for always known set of methods. It could be coded in explicitly:
https://github.com/Cornices/cornice/blob/master/cornice/service.py#L208

### PR DESCRIPTION
**Pull request by fizyk**

### CODE REVIEW

File: tests/test_service.py [hunk 1]
```diff
@@ -198,6 +198,27 @@ def patch_favorite_color(request):
...
 
+        @service.put()
+        def put_favoruite_color(request):
+            return ""
+
+        method, view, _ = service.definitions[5]
+        self.assertEqual("PUT", method)
+
+        @service.delete()
+        def delete_favoruite_color(request):
```
Comment (Natim) [inline tests/test_service.py:209]:
```suggestion
        def delete_favorite_color(request):
```

File: tests/test_service.py [hunk 2]
```diff
@@ -198,6 +198,27 @@ def patch_favorite_color(request):
         method, view, _ = service.definitions[4]
         self.assertEqual("PATCH", method)
 
+        @service.put()
+        def put_favoruite_color(request):
```
Comment (Natim) [inline tests/test_service.py:202]:
```suggestion
        def put_favorite_color(request):
```

File: tests/test_service.py [hunk 3]
```diff
@@ -198,6 +198,27 @@ def patch_favorite_color(request):
...
+
+        @service.delete()
+        def delete_favoruite_color(request):
+            return ""
+
+        method, view, _ = service.definitions[6]
+        self.assertEqual("DELETE", method)
+
+        @service.options()
+        def favoruite_color_options(request):
```
Comment (Natim) [inline tests/test_service.py:216]:
```suggestion
        def favorite_color_options(request):
```

File: .gitignore [hunk 4]
```diff
@@ -20,3 +20,4 @@ man
 .channel
 dist/
 .venv/
+.idea/
```
Comment (Natim) [inline .gitignore:23]: Leftover?

Comment (fizyk) [reply .gitignore:23]: Should I remove it? It's for the IntelliJ/PyCharm project files I don't want to commit into repository

Comment (Natim) [reply .gitignore:23]: No I guess we can keep it, I wonder why it is not possible to configure PyCharm to not put file in the repository but rather un `~/.idea/project_name`

**Comment by fizyk** [comment]
@Natim code updated.

**Pull request merged**